src/vision/status_quo/barbara_writes_a_runtime_agnostic_lib.md

```diff
@@ -0,0 +1,125 @@
+# 😱 Status quo stories: Barbara writes a runtime-agnostic library
+
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from
+real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async
+Rust programmers face today.
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR
+making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories
+[cannot be wrong], only inaccurate). Alternatively, you may wish to
+[add your own status quo story][htvsq]!
+
+## The story
+
+Barbara and Alan work at AmoolgeSoft, where many teams are switching from Java to Rust. These teams
+have many different use cases and various adoption stories. Some teams are happy users of tokio,
+others happy users of async-std, and others still are using custom runtimes for highly specialized
+use cases.
+
+Barbara is tasked with writing a library for a custom protocol, SLOW (only in use at AmoogleSoft)
+and enlists the help of Alan in doing so. Alan is already aware that [not all libraries in Rust work
+with all runtimes][nalirwwar]. Alan and Barbara start by writing a parser which works on
+`std::io::Read` and get their tests working with `String`s. After this they contemplate the question
+of how to accept a TCP connection.
+
+### Incompatible `AsyncRead` traits
+
+Alan asks Barbara what is the async equivalent is of `std::io::Read`, and Barbara sighs and says
+that there isn't one. Barbara brings up tokio's and the [futures crate]'s versions of `AsyncRead`.
+Barbara decides not to talk about `AsyncBufRead` for now.
+
+Barbara and Alan decide to use the future's `AsyncRead` for no other reason other than it is
+runtime-agnostic. Barbara tells Alan not to worry as they can translate between the two. With some
+effort (**TODO:** maybe link a story about `Pin`? which one?) they convert their parser to using
```

Comment:
I think we don't have a good story for `AsyncRead` yet. We do mention `Pin` difficulties in a few places like in [here](https://rust-lang.github.io/wg-async-foundations/vision/status_quo/barbara_needs_async_helpers.html) and [here](https://github.com/rust-lang/wg-async-foundations/blob/master/src/vision/status_quo/alan_hates_writing_a_stream.md), but as we discussed today in the session on `AsyncRead`/`AsyncWrite`, we'll probably want a dedicated story for this.

Reply:
This one was about Pin difficulties specifically so I've added links to those stories now (thanks).